### ISSUE TITLE
Allow `nuxt/nuxt-config-keys-order` rule to be used on files chosen by the user

### ISSUE THREAD

**Issue by gt-hive-benedict-lee** [nuxt/eslint#606]
### Describe the feature

Currently, `nuxt/nuxt-config-keys-order` only works on files named exactly as `nuxt.config.{ts,js,mts,mjs,tsx,jsx}`. As my Nuxt config has grown larger, I am considering separating out the environment configs into their own separate files. In these separate files, I would like to be able to apply the same sort rule that is applied to my main Nuxt config.

However, it is currently impossible to make ESLint apply this rule to those files,  as due to [these lines](https://github.com/nuxt/eslint/blob/main/packages/eslint-plugin/src/rules/nuxt-config-keys-order/index.ts#L24-L26) processing is immediately stopped on any other file name, and so even if I configure ESLint to apply `nuxt/nuxt-config-keys-order` to my extracted out environment config file, no errors are given by ESLint.

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [x] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [x] Check existing [discussions](https://github.com/nuxt/eslint/discussions) and [issues](https://github.com/nuxt/eslint/issues).

**Comment by benedictleejh** [comment]
Some experimenting suggests that removing the lines https://github.com/nuxt/eslint/blob/main/packages/eslint-plugin/src/rules/nuxt-config-keys-order/index.ts#L24-L26 is enough to get this to work, and because of the default files glob set in https://github.com/nuxt/eslint/blob/main/packages/eslint-config/src/configs/nuxt.ts, it still works normally with the root `nuxt.config.ts`.

Would this be an acceptable implementation? If so, I can submit a PR for it.

**Comment by kilisei** [comment]
i used 2 patches to get it to work

i use `"@nuxt/eslint": "^1.9.0"`

@nuxt/eslint-plugin:
```patch
diff --git a/dist/index.mjs b/dist/index.mjs
index db5bee80d5b4053770b679718379b159a5193ec9..c835bc7a91b7228fcde3992cc06d912326c844f3 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -158,9 +158,6 @@ const rule = createRule({
   },
   defaultOptions: [],
   create(context) {
-    if (!context.filename.match(/nuxt\.config\.[mc]?[jt]sx?$/)) {
-      return {};
-    }
     return {
       ExportDefaultDeclaration(node) {
         let object;

```

@nuxt/eslint-config:
```patch
diff --git a/dist/shared/eslint-config.6zQe2jV2.mjs b/dist/shared/eslint-config.6zQe2jV2.mjs
index 00e92bb9d7bb34bc8b46bcdbb355cb48e6446c7a..2fc150f9faa0ca654626c371c062b1dacb8060eb 100644
--- a/dist/shared/eslint-config.6zQe2jV2.mjs
+++ b/dist/shared/eslint-config.6zQe2jV2.mjs
@@ -153,10 +153,11 @@ function nuxt(options) {
       "nuxt/prefer-import-meta": "error"
     }
   });
-  if (sortConfigKeys) {
+  if (sortConfigKeys) {
     configs.push({
       name: "nuxt/sort-config",
       files: [
+        "**/.config/nuxt.?([cm])[jt]s?(x)",
         "**/nuxt.config.?([cm])[jt]s?(x)"
       ],
       rules: {

```

**Comment by antfu** [comment]
Would you like to send a PR to fix it? Thanks
